### PR TITLE
Agent release management trigger are now following the triggered pipeline progress

### PR DESCRIPTION
**Pull request by Pythyu**
### What does this PR do?

Allow agent release management trigger to follow the pipeline to fail or succeed depending of the downstream pipeline status.

### Motivation

Currently with a full green pipeline, it's not assured that the package deployment is successful.

### Additional Notes

Depends on this [PR](https://github.com/DataDog/agent-release-management/pull/125) in agent-release-management
will remove the custom branch `pythyu/waiting_trigger` once the above PR is merged

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by KevinFairise2** [comment]
**🔨 warning:** ‏
The use of `follow` here will make the job wait for the completion of the pipeline created on `agent-release-management`: https://gitlab.ddbuild.io/DataDog/agent-release-management/-/pipelines/23573360
But it will not wait for the child pipeline triggered by this one. And so will not wait for the pipeline with all the jobs: https://gitlab.ddbuild.io/DataDog/agent-release-management/-/pipelines/23573503

File: .gitlab/trigger_release.yml [hunk 1]
```diff
@@ -20,7 +20,8 @@
     # when triggered with major version 7
     - source /root/.bashrc
     - export RELEASE_VERSION=$(inv agent.version --major-version 7 --url-safe --omnibus-format)-1
-    - inv pipeline.trigger-child-pipeline --no-follow --project-name "DataDog/agent-release-management" --git-ref "main" --variables "ACTION,AUTO_RELEASE,BUILD_PIPELINE_ID,RELEASE_PRODUCT,RELEASE_VERSION,TARGET_REPO,TARGET_REPO_BRANCH"
+    - export GITLAB_TOKEN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.gitlab_pipelines_scheduler_token --with-decryption --query "Parameter.Value" --out text)
```
Comment (chouquette) [inline .gitlab/trigger_release.yml:23]: Why is the token required now? It appears to be the only change in this PR but the trigger was working before without it no?

Comment (Pythyu) [reply .gitlab/trigger_release.yml:23]: To follow the pipeline in a different repository we need to have the bot token

Comment (chouquette) [reply .gitlab/trigger_release.yml:23]: oh? I thought it worked without, my bad
LGTM then!

**Review by chouquette — APPROVED** [review]

File: .gitlab/trigger_release.yml [hunk 2]
```diff
@@ -20,7 +20,8 @@
...
+    - inv pipeline.trigger-child-pipeline --project-name "DataDog/agent-release-management" --git-ref "pythyu/waiting_trigger" --variables "ACTION,AUTO_RELEASE,BUILD_PIPELINE_ID,RELEASE_PRODUCT,RELEASE_VERSION,TARGET_REPO,TARGET_REPO_BRANCH"
```
Comment (chouquette) [inline .gitlab/trigger_release.yml:24]: Just a friendly reminder to update this!

**Pull request merged**